tests/integration_tests/functional/test_cpu_features.py

```diff
@@ -146,6 +146,35 @@ def test_cpu_template(test_microvm_with_ssh, network_config, cpu_template):
     containing all features on a t2/c3 instance and check that the cpuid in
     the guest is an exact match of the template.
     """
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+
+    test_microvm.basic_config(vcpu_count=1)
+    # Set template as specified in the `cpu_template` parameter.
+    response = test_microvm.machine_cfg.put(
+        vcpu_count=1,
+        mem_size_mib=256,
+        ht_enabled=False,
+        cpu_template=cpu_template,
+    )
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+    _tap, _, _ = test_microvm.ssh_network_config(network_config, '1')
+
+    response = test_microvm.actions.put(action_type='InstanceStart')
+    if utils.get_cpu_vendor() != utils.CpuVendor.INTEL:
+        # We shouldn't be able to apply Intel templates on AMD hosts
+        assert test_microvm.api_session.is_status_bad_request(
+            response.status_code)
+        return
+
+    assert test_microvm.api_session.is_status_no_content(
+            response.status_code)
+    masked_features(test_microvm, cpu_template)
+    enabled_features(test_microvm, cpu_template)
```

Comment:
Nit: I would rename this to `check_masked_features`, `check_enabled_features` or something similar.

Reply:
done.